doc/source/glossary.rst

```diff
@@ -0,0 +1,188 @@
+    Pages, like pages in a book, make up the whole, but allow for specific chunks to be allocated and used as necessary.
+    Operating systems uses pages as a means to have granular control over chunks of memory.  This allows them to be
+    reordered and reused as necessary (without having to move large chunks of data around), and allows them to have
+    access controls placed upon them, limiting actions such as reading and writing.
+
+.. _Page Table:
+
+Page Table
+    A table that points to a series of :ref:`pages<page>`.  Each page table is typically the size of a single page,
+    and page tables can point to pages that are in fact other page tables.  Using tables that point to tables, it's
+    possible to use them as a way to map a particular address within a (potentially larger, but sparsely populated)
+    virtual space to a concrete (and usually contiguous) physical space, through the process of :ref:`mapping<map>`.
+
+.. _Pointer:
+
+Pointer
+    A value within memory that points to a different area of memory.  This allows objects to contain references to
+    other objects without containing all the data of the other object.  Following a pointer is known as :ref:`dereferencing<dereference>`
+    a pointer.  Pointers are usually the same length as the maximum address of the address space, since they
+    should be able to point to any address within the space.
+
+R
+-
+.. _Range:
+
+Range
+    This is the set of the possible output values for a mapping or mathematical function.
+
+S
+-
+.. _Struct:
+
+Struct, Structure
+    A means of containing multiple different :ref:`type<types>` associated together.  A struct typically contains
+    other :ref:`type<types>`, usually :ref:`aligned<alignment>` (unless :ref:`packing<packed>` is involved).  In this way
+    the :ref:`members<member>` of a type can be accessed by finding the data at the relative :ref:`offset<offset>` to
+    the start of the structure.
+
+.. _Symbol:
+
+Symbol
+    This is used in many different contexts, as a short term for many things.  Within Volatility, a symbol is a
+    construct that usually encompasses a specific type :ref:`type<Type>` at a specfific :ref:`offset<Offset>`,
+    representing a particular instance of that type within the memory of a compiled and running program.  An example
+    would be the location in memory of a list of active tcp endpoints maintained by the networking stack
+    within an operating system.
+
+T
+-
+.. _Template:
+
+Template
+    Within volatility 3, the term template applies to a :ref:`type<Type>` that has not yet been instantiated or linked
+    to any data or a specific location within memory.  Once a type has been tied to a particular chunk of data, it is
+    called an :ref:`object<Object>`.
+
+.. _Translation Layer:
+
+Translation Layer
+    This is a specific type of :ref:`data layer<Data Layer>`, a non-contiguous group of bytes that can be references by
```

Comment:
Yeah, I like that, the reason it's linear is because in general `<size of function input> == <size of function output>` which isn't true for compression (for example).  With compression a chunk of 1024 bytes might map to a region that's say 500 bytes, and that's what I class as non-linear.  If there's a better term for it, I'm happy to go for it.  Generally `translate(input) + c = translate(input + c)` assuming `c` stays within the contiguous block that `input` is a part of (for example a page). I'm not sure an Associative Layer would give the right impression on hearing it?  5:S